pkg/controller/storagecluster/reconcile.go

```diff
@@ -739,23 +738,8 @@ func newCephCluster(sc *ocsv1.StorageCluster, cephImage string) *cephv1.CephClus
 		},
 	}
 
-	// If a MonPVCTemplate is provided, use that. If not, if StorageDeviceSets
-	// have been provided, use the StorageClass of the DataPVCTemplate from the
-	// first StorageDeviceSet for providing the Mon PVs
 	if sc.Spec.MonPVCTemplate != nil {
 		cephCluster.Spec.Mon.VolumeClaimTemplate = sc.Spec.MonPVCTemplate
-	} else if len(sc.Spec.StorageDeviceSets) > 0 {
```

Comment:
In addition to removing use of the StorageDeviceSet, we should expose the dataDirHostPath setting from the StorageCluster CR. We can't always assume `/var/lib/rook` is the right value for `dataDirHostPath` as seen above on line 714.

Reply:
yeah will add this

Reply:
done